Sources/NIO/Utilities.swift

```diff
@@ -41,7 +52,37 @@ public enum System {
     ///
     /// - returns: The logical core count on the system.
     public static var coreCount: Int {
+#if os(Windows)
+        var dwLength: DWORD = 0
+        _ = GetLogicalProcessorInformation(nil, &dwLength)
+
+        let alignment: Int =
+            MemoryLayout<SYSTEM_LOGICAL_PROCESSOR_INFORMATION>.alignment
+        let pBuffer: UnsafeMutableRawPointer =
+            UnsafeMutableRawPointer.allocate(byteCount: Int(dwLength),
+                                             alignment: alignment)
+        defer {
+            pBuffer.deallocate()
+        }
+
+        let dwSLPICount: Int =
+            Int(dwLength) / MemoryLayout<SYSTEM_LOGICAL_PROCESSOR_INFORMATION>.size
```

Comment:
I think we have to use `.stride` here, not `.size`, as we're trying to treat this as an array.

(Apropos of nothing it's super annoying that `GetLogicalProcessorInformation` returns the size of the buffer in _bytes_ instead of in element count.

Reply:
Hmm, MSDN agrees about the size.  The returned value doesn’t talk about the stride, it is talking about size.  Consider tail allocations or alignment which would not be included.  This is not swift-esque, this is C more or less.  I think that size is much more clear.

Reply:
@compnerd MSDN refers to C's `sizeof` which gives you the _stride_. In Swift, you definitely have to use `.stride` and not `.size`.

Reply:
+1. This is also entirely natural when you consider that the return value _must_ be aligned to use safely, and so this must be a standard C-style array, which lays out objects based on their stride, not their size.

Reply:
Swift:
```
$ cat test.swift
struct a {
    var x: CInt
    var y: CChar
}

print(MemoryLayout<a>.size, MemoryLayout<a>.stride)
$ swift test.swift
5 8
```

C:
```
#include <stdio.h>

struct a {
    int x;
    char y;
};

int main() {
    printf("%lu\n", sizeof(struct a));
    return 0;
}
$ clang test.c && ./a.out
8
```

Reply:
Though it should be noted that on at least some versions of Swift it seems that Swift "knows" that a struct came from C and just lies. Consider this:

---
`lib.h`

```C
#ifndef LIB_H
#define LIB_H

struct LibStruct {
    int foo;
    char bar;
};

int szof_lib(void);

#endif
```

---

`lib.c`

```C
#include "lib.h"

int szof_lib(void) {
    return sizeof(struct LibStruct);
}
```

---

`main.swift`

```swift
import lib

print(MemoryLayout<LibStruct>.size)
print(MemoryLayout<LibStruct>.stride)
print(szof_lib())
```

---

The output of this program is:

```
8
8
8
```

Reply:
yeah, it's the padding from C...

Reply:
C adds padding (which Swift imports correctly) to make stride == size. Still, in Swift `.stride` is correct :)

Reply:
you can see it with `-Wpadded`:

```
$ clang -Wpadded test.c && ./a.out 
test.c:5:8: warning: padding size of 'struct a' with 3 bytes to alignment boundary [-Wpadded]
struct a {
       ^
1 warning generated.
8, 4
```

Reply:
Oh sure, I agree that `.stride` is right, it just so happens that in this case `.size` was going to get the right outcome anyway.

Reply:
but why not use `.stride` which works everywhere in Swift? If you use `.size`, then you need the programmer who reads this to understand that for C structs `.size == .stride` and that your struct is in fact imported from C, ...

Reply:
Because I've found that knowing where the struct is defined has been extremely useful in porting - there are differences in the generated code patterns and generally the debugging technique that works for swift currently is print or read disassembly.